source/main/zettlr.ts

```diff
@@ -149,9 +154,11 @@ export default class Zettlr {
     // listen to close-events on the main window, we should be able to handle
     // this, if we ever switched to the auto updater.
     app.on('before-quit', (event) => {
+      this.isQuiting = true
       if (!this._documentManager.isClean()) {
         // Immediately prevent quitting ...
         event.preventDefault()
+        this.isQuiting = false
```

Comment:
What is the logic of this variable, exactly? I'm just asking because I see what you want to achieve, but I'm curious if that's not possible with the state of the application itself …?

Reply:
Can quit Zettlr by either:

1. titlebar `x`; or
2. `File` → `Quit`; `Ctrl-Q`; Tray → `Quit` etc.

`isQuiting` is set true when **2** occurs. Therefore do not prevent quitting when **2** occurs, see [index.ts:L279](https://github.com/Zettlr/Zettlr/pull/2026/files#diff-87c7699f8e2c0214b588787d0bb57543927f3f8857444550bf0dbb35d5294c06R279)

I tried other methods, but the other methods behave identically for **1** and **2**. The Electron | [app](https://www.electronjs.org/docs/api/app) API doesn't provide any state to help.

I will add a code comment to clarify this. Are you ok with the `isQuitting` implementation @nathanlesage ?

I will fix the misspelling.